ql/src/java/org/apache/hadoop/hive/ql/exec/ddl/database/AlterDatabaseDesc.java

```diff
@@ -16,112 +16,100 @@
  * limitations under the License.
  */
 
-package org.apache.hadoop.hive.ql.plan;
+package org.apache.hadoop.hive.ql.exec.ddl.database;
 
 import java.io.Serializable;
 import java.util.Map;
 
+import org.apache.hadoop.hive.ql.exec.ddl.DDLDesc;
+import org.apache.hadoop.hive.ql.exec.ddl.DDLTask2;
 import org.apache.hadoop.hive.ql.parse.ReplicationSpec;
+import org.apache.hadoop.hive.ql.plan.Explain;
 import org.apache.hadoop.hive.ql.plan.Explain.Level;
+import org.apache.hadoop.hive.ql.plan.PrincipalDesc;
 
 /**
- * AlterDatabaseDesc.
- *
+ * DDL task description for ALTER DATABASE commands.
  */
 @Explain(displayName = "Alter Database", explainLevels = { Level.USER, Level.DEFAULT, Level.EXTENDED })
 public class AlterDatabaseDesc extends DDLDesc implements Serializable {
-
   private static final long serialVersionUID = 1L;
 
-  // Only altering the database property and owner is currently supported
-  public static enum ALTER_DB_TYPES {
-    ALTER_PROPERTY, ALTER_OWNER, ALTER_LOCATION
-  };
-
-  ALTER_DB_TYPES alterType;
-  String databaseName;
-  Map<String, String> dbProperties;
-  PrincipalDesc ownerPrincipal;
-  ReplicationSpec replicationSpec;
-  String location;
+  static {
+    DDLTask2.registerOperator(AlterDatabaseDesc.class, AlterDatabaseOperation.class);
+  }
 
   /**
-   * For serialization only.
+   * Supported type of alter db commands.
+   * Only altering the database property and owner is currently supported
    */
-  public AlterDatabaseDesc() {
-  }
+  public enum AlterDbType {
+    ALTER_PROPERTY, ALTER_OWNER, ALTER_LOCATION
+  };
 
-  public AlterDatabaseDesc(String databaseName, Map<String, String> dbProps,
```

Comment:
This would result in having longer classes, and putting one indentation level lower everything. Also later I'd consider moving the related parts of the DDLSemanticAnalyzer next to these classes in a third class, which if put into the same class would also result even longer container classes. I believe having a separate class for Desc, Operation and Analyzer is cleaner.

Reply:
I don't feel a few hundred lines much :) tought that closely related stuff might be beneficial during future feature additions/etc
but it was just an idea; it will be fine without it as well.

I'm right now wondering about what would happen if instead of adding an extra layer of  (`AlterDatabaseOperation`,`AlterDatabaseDesc`) ; how insane would be to do: `AlterDatabaseTask`,`Work` and `SA`?

Reply:
Basically AlterDatabaseOperation (or in more general every XXXOperation could be a task as well, and every XXXDesc could have their own XXXWork class. But that would mean a lot of repetitive code, so having a common Task, and Work class for them is just reusing the same boilerplate code. Having a separate SA is a good idea, I think when DDLTask is cut to pieces the next step should be a to cut DDLSemanticAnalyzer too!

Reply:
I see; so you've already considered considered and rejected this approach :)